### PR TITLE
Support filtering out warnings at the file level.

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md has been updated
 - [x] Blocked on https://github.com/Polymer/polymer-project-config/pull/45 being released

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-editor-service/94)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
   `lint.ignoreWarnings` which takes an array of warning codes that we should
   not report. This brings the IDE in line with `polymer lint` on the command
   line.
+- Supports specifying files to ignore all warnings for with the polymer.json
```
Comment (valdrinkoshi) [inline CHANGELOG.md]: nit: format `polymer.json`

File: src/language-server/diagnostics.ts [hunk 2]
```diff
@@ -117,18 +121,28 @@ export default class DiagnosticGenerator extends Handler {
...
+        try {
+          rules = registry.getRules(projectConfig.lint.rules);
+        } catch (e) {
+          // TODO(rictic): let the user know about this error, and about
+          //   this.settings.projectConfigDiagnostic if it exists.
+        }
+      }
+      if (lintConfig.ignoreWarnings) {
+        this.warningCodesToFilterOut =
+            new Set(projectConfig.lint.ignoreWarnings);
```
Comment (valdrinkoshi) [inline src/language-server/diagnostics.ts]: nit: use the shorter `lintConfig.ignoreWarnings`. You could avoid the `if/else` check and just set 
```ts
this.warningCodesToFilterOut = new Set(lintConfig.ignoreWarnings);

File: src/language-server/diagnostics.ts [hunk 3]
```diff
@@ -117,18 +121,28 @@ export default class DiagnosticGenerator extends Handler {
...
-      try {
-        rules = registry.getRules(projectConfig.lint.rules);
-      } catch (e) {
-        // TODO(rictic): let the user know about this error, and about
-        //   this.settings.projectConfigDiagnostic if it exists.
+    if (projectConfig.lint) {
+      const lintConfig = projectConfig.lint;
+      if (lintConfig.rules) {
+        try {
+          rules = registry.getRules(projectConfig.lint.rules);
```
Comment (valdrinkoshi) [inline src/language-server/diagnostics.ts]: nit: use the shorter `lintConfig.rules`

File: src/language-server/diagnostics.ts [hunk 4]
```diff
@@ -146,14 +160,16 @@ export default class DiagnosticGenerator extends Handler {
       const warnings = await this.linter.lintPackage();
       this.reportPackageWarnings(warnings);
     } else {
-      const warnings = await this.linter.lint(this.documents.keys().map(
-          uri => this.converter.getWorkspacePathToFile({uri})));
+      const paths =
+          this.documents.keys()
```
Comment (valdrinkoshi) [inline src/language-server/diagnostics.ts]: mini mini mini optimization: keep the documents keys in a constant to reuse it for `paths` and `diagnosticsByUri`

File: src/language-server/diagnostics.ts [hunk 5]
```diff
@@ -170,6 +186,16 @@ export default class DiagnosticGenerator extends Handler {
     }
   }
 
+  private filterWarnings(warnings: ReadonlyArray<Warning>):
+      ReadonlyArray<Warning> {
+    return warnings.filter(w => {
+      return !(
```
Comment (valdrinkoshi) [inline src/language-server/diagnostics.ts]: nit: you could remove this `return` and inline to
```js
return warnings.filter(w => !(

File: src/test/diagnostics_test.ts [hunk 6]
```diff
@@ -183,6 +183,27 @@ suite('DiagnosticGenerator', function() {
         ['unknown-superclass']);
   });
 
+  test('can be configured to filter out some files', async() => {
+    const {client} = await createTestEnvironment();
+    await client.openFile('index.html', `
+      <link rel="import" href="nonexistant.html">
+      <script>
+        class Foo extends What {};
```
Comment (valdrinkoshi) [inline src/test/diagnostics_test.ts]: class Foo extends WOOT? 😁

**Comment by rictic** [comment]
Good suggestions! Cleaned up

**Pull request merged**